File: api/src/main/java/myapp/user/management/service/UserManagementService.java
package myapp.user.management.service;

import myapp.user.management.service.bean.ChangeStatusRequest;
import myapp.user.management.service.bean.GetUserListServiceReply;
import myapp.user.management.service.bean.UserDeleteServiceRequest;
import myapp.user.management.service.bean.UserEditServiceRequest;

/**
 * CustomerManagementService API
 * 
 * @author Mate
 *
 */
public interface UserManagementService
{
	/**
	 * Method for retrieving users
	 * 
	 * @return {@link GetUserListServiceReply}
	 */
	GetUserListServiceReply getUserList();

	/**
	 * Method for deleting user by Id
	 * 
	 * @param request {@link UserDeleteServiceRequest}
	 */
	void deleteUserById(UserDeleteServiceRequest request);

	/**
	 * Method for editing user
	 * 
	 * @param request {@link UserEditServiceRequest}
	 */
	void editUser(UserEditServiceRequest request);

	/**
	 * Method for changing account status of user
	 * 
	 * @param request {@link ChangeStatusRequest}
	 */
	void changeAccountStatus(ChangeStatusRequest request);
}

File: web/src/main/webapp/resources/js/myapp/admin/usermanagement/UserManagementService.js
'use strict';

function UserManagementService($http) {

	var service = {
		fetchAllUsers : fetchAllUsers,
		createUser : createUser,
		deleteUser : deleteUser,
		editUser : editUser,
		changeAccountStatus : changeAccountStatus
	}
	return service;
	
	function fetchAllUsers() {
		return $http.post('usermanagement/getUsers');
	}
	
	function changeAccountStatus(status, username) {
		var changeStatusRequest = {
			status: status,
			username: username
		};
		return $http.post('usermanagement/changeAccountStatus', changeStatusRequest);
	}
	
	function createUser(data) {
		var request = {
			customer: data
		};
		return $http.post('usermanagement/addUser', request);
	}
	
	function deleteUser(customerId) {
		var request = {
			id : customerId
		};
		return $http.post('usermanagement/deleteUser', request);
	}
	
	function editUser(user) {
		var request = {
			user: user
		};
		return $http.post('usermanagement/editUser', request);
	}

};